docs/concepts/ipld.md

```diff
@@ -11,4 +11,98 @@ related:
+│                 1. Data model layer              │
+│  (basic types for single-block data structures)  │
+│                                                  │
+├──────────────────────────────────────────────────┤
+│                                                  │
+│                   0. Block layer                 │
+│               (CID, data, codec)                 │
+│                                                  │
+└──────────────────────────────────────────────────┘
+```
+
+## Example: Path traversal
+
+This example will create a simple link from a node to a Git commit. The example will use the CLI, but this can be done in many other ways (for example, using [JavaScript](https://github.com/ipfs/js-ipfs-http-client)).
+
+1. Pick a Git CID. This example will use [explorer.ipld.io](https://explore.ipld.io/#/explore/z8mWaJHXieAVxxLagBpdaNWFEBKVWmMiE). The goal is to create a node that links to this CID:
+
+```
+z8mWaJHXieAVxxLagBpdaNWFEBKVWmMiE
+```
+
+2. Start the IPFS daemon.
+
+```bash
+$ ipfs daemon
+```
+
+2. Create some data. Notice how this data stores the Git CID.
+
+```bash
+$ echo '{ "message": "linking to git commit", "files": [{ "/": "z8mWaJHXieAVxxLagBpdaNWFEBKVWmMiE" }] }' > test.json
+$ cat test.json
+{ "message": "linking to git commit", "files": [{ "/": "z8mWaJHXieAVxxLagBpdaNWFEBKVWmMiE" }] }}
+```
+
+3. Store an IPLD format node using the [put](https://docs.ipfs.io/reference/api/cli/#ipfs-dag-put) command.
+
+```bash
+$ ipfs dag put test.json
+bafyreifsnd635mi5ratse2mzg3wget4afcwf2tsop4lymbxtj27tfgizia
+```
+
+4. From here, use the CID provided to get an IPLD-format node.
+
+```bash
+$ ipfs dag get bafyreifsnd635mi5ratse2mzg3wget4afcwf2tsop4lymbxtj27tfgizia
+{"files":[{"/":"baf4bcfg4ep767tjp5lxyanx5urpjjgx5q2volvy"}],"message":"linking to git commit"}
+```
+
+5. Finally, traverse through to access the Git commit.
+
+```bash
+$ ipfs dag get bafyreifsnd635mi5ratse2mzg3wget4afcwf2tsop4lymbxtj27tfgizia/files/0/
+{"author":{"date":"1524708555 +0200","email":"magik6k@users.noreply.github.com","name":"Łukasz Magiera"},"committer":{"date":"1524708555 +0200","email":"noreply@github.com","name":"GitHub"},"message":"Merge pull request #31 from ipfs-shipyard/feat/zip-button\n\nDownload Zip button","other":[" "],"parents":[{"/":"baf4bcfcjvhadrgzmwjflz25kyyc3tolmqaobvfy"},{"/":"baf4bcfcfa2xemiceossebsosuxsi77m5ztatwci"}],"signature":{"Text":" \n wsBcBAABCAAQBQJa4TTLCRBK7hj4Ov3rIwAAdHIIAFIOYsbNvoXO+N7d1pnhR6Ue\n qE2qLrldXeGI+04k3i0rmL8/YAi3zavCA1F1VfjSxX8nYsdYWmjljURiDL07y+5K\n RDxsdHPvf8GUN04rO3mVbCISAme3okGMdrn+hfEdZVdwGdUFva0vtegMWDvnADnY\n si6kURUTONywkNLLHsUD6kPUk2YEZuruaSWblGQwS1ejGzfPo7n4z/ScotC/XxFP\n SCj93XLLlh6HBhh4huJgr2HA/13Tg4XLW5+DnE1mIs3FA0cCQ2WGZCbXEugX/+xv\n u0UWyVv7o8xWt0SJeAFpOLDy7MhCpETJsQ0sFe6Mte6plzsa4vM8/Rq+nI5TzoU=\n =RYjo\n"},"tree":{"/":"baf4bcfcqex3wt5ajfg3lynpufxpiivqt5nxlmxy"}}"
+```
+
+Congratulations! We successfuly linked a node with CID `bafyreifsnd...` to `z8mWaJHXieA...`.
```

Comment:
Can proto.school look at the example? I'd like it to live somewhere.

https://github.com/ipfs/docs/issues/86 wanted some video/code example. Going through this helped me put the pieces together. Especially the key value of "/". At first I was using a different key value "foobar" so my traverse was:

{ "message": "linking to git commit", "files": [{ "foobar": "z8mWaJHXieAVxxLagBpdaNWFEBKVWmMiE" }] }}

<my-cid>/files/0/foobar
returns git-cid
use git-cid to continue traverse
<git-cid>/author...
Not very elegant.

Changing to "/" made the traverse much nicer. 
{ "message": "linking to git commit", "files": [{ "/": "z8mWaJHXieAVxxLagBpdaNWFEBKVWmMiE" }] }}
<my-cid>/files/0/author

Reply:
@leon-do @hsanjuan -- You both make good points. To that end, what about the following approach?

1. Please explain in text @hsanjuan's point about "how we actually know that something is a Git CID", so that's conceptually covered in the main body text of the doc.

2. Keep the code example, but don't make doing the code example essential to one's understanding of the topic (acid test: what if you were just reading this on your phone on the bus, and couldn't play with code?). One way to do this is to annotate/comment or contextualize *why* the code example is helpful (which you do in your last comment above about putting the pieces together). Another is to relocate it within the content itself so it's closer to the end and a bit more like an appendix: if the reader's eyes glaze over upon seeing a code block and they don't go any further, they should still be able to have enough information to understand the main context.

In the long term, agree that a ProtoSchool tutorial would be awesome -- but developing these is a little more involved, and it would be sad to lose momentum or defer presenting useful information in the meantime. (FYI: Guidelines on building ProtoSchool tutorials are here: https://github.com/ProtoSchool/protoschool.github.io )

Thanks both!

Reply:
Sounds good to me @jessicaschilling 

Thoughts @hsanjuan ?

FYI: I'll be happy to write some code for protoschool. I'll have to familiarize myself with the repo first.

Reply:
My main issue with code examples is that they break silently. The last thing we'll remember as developers is to come fix an example embedded in some documentation page, particularly if we have many. But this is not a problem with the example itself, but with how bad we are at doc maintenance. There are other ways to fix it (a big button "Does it not work? Report a problem with this example" perhaps). If @jessicaschilling wants the example, then that's good, let's keep it.

Reply:
Agreed about the code example, but also agreed on the opportunity to give feedback. We have the following on the bottom of every page and it's already caught a few problems:
![image](https://user-images.githubusercontent.com/1507828/73681537-c77cfb80-467b-11ea-847d-35d4ded92162.png)
So (for now at least) I'd recommend keeping the code example, downplayed as above. Thanks!

Reply:
@leon-do PS - @terichadbourne is the ProtoSchool lead and would love to talk to you if you're interested in contributing lesson material!

Reply:
At first glance it looks to me like these commands to use the DAG API within IPFS are already covered in ProtoSchool. Please see https://proto.school/#/basics (introducing the basic methods) and https://proto.school/#/blog (more complex example of linking data) and let me know if you disagree. If we all agree these are covering the same concepts, let's be sure to link to those tutorials from the docs for further learning. Thanks!

Reply:
Hello Teri,

Read https://proto.school/#/basics/03
That covers the example. I'll link to it instead!